### PR TITLE
Fix bug that release attachment files not deleted when deleting repository

### PR DESCRIPTION
**Pull request by lunny**
As title.

### CODE REVIEW

**Comment by lunny** [comment]
Should fix #9321

File: models/repo.go [hunk 1]
```diff
@@ -2129,6 +2135,23 @@ func DeleteRepository(doer *User, uid, repoID int64) error {
...
+	// We should always delete the files after the database transaction succeed. If
+	// we delete the file but the database rollback, the repository will be borken.
+
+	// Remove issue attachment files.
+	for i := range attachmentPaths {
+		removeAllWithNotice(x, "Delete issue attachment", attachmentPaths[i])
+	}
+
+	// Remove release attachment files.
+	if len(releaseAttachments) > 0 {
```
Comment (sapk) [inline models/repo.go]: The length check before the loop seems not needed

**Review by sapk — COMMENTED** [review]
looks good but maybe remove un-needed check

**Review by axifive — APPROVED** [review]

File: models/repo.go [hunk 2]
```diff
@@ -1992,6 +1992,17 @@ func DeleteRepository(doer *User, uid, repoID int64) error {
 		}
 	}
 
+	releaseAttachments := make([]string, 0, 20)
+	attachments := make([]*Attachment, 0, len(releaseAttachments))
```
Comment (guillep2k) [inline models/repo.go]:
```suggestion
	attachments := make([]*Attachment, 0, cap(releaseAttachments))
```
Or better, move this to after `releaseAttachments ` is populated.

File: models/repo.go [hunk 3]
```diff
@@ -2043,7 +2054,7 @@ func DeleteRepository(doer *User, uid, repoID int64) error {
 	}
 
 	attachmentPaths := make([]string, 0, 20)
-	attachments := make([]*Attachment, 0, len(attachmentPaths))
+	attachments = make([]*Attachment, 0, len(attachmentPaths))
```
Comment (guillep2k) [inline models/repo.go]: Why do we need to get the attachment list again?

Comment (lunny) [reply models/repo.go]: The `attachment` table stored issue/comment attachments and release attachments. Second is get issue/comment attachments list.

Comment (axifive) [inline models/repo.go]: Maybe the second `make()` isn't needed?
```suggestion
	attachments = attachments[:0]
```

File: models/repo.go (hunk 2 above)
Comment (sapk) [reply models/repo.go]: @guillep2k attachments is populated by xorm before releaseAttachments. 
Maybe you mean :
```
attachments := make([]*Attachment, 0, 20)
if err = sess.Join("INNER", "release", "release.id = attachment.release_id") ... {
...
}
releaseAttachments := make([]string, len(attachments))
for i := 0; i < len(attachments); i++ {
	releaseAttachments = append(releaseAttachments, attachments[i].LocalPath())
}
```

File: models/repo.go (hunk 3 above)
Comment (sapk) [reply models/repo.go]: @axifive @guillep2k  Thoses improvements are maybe not related to the fix and could be done outside to limit backport ?

**Review by sapk — APPROVED** [review]

File: models/repo.go (hunk 3 above)
Comment (axifive) [reply models/repo.go]: @sapk, These are very minor corrections that don't change the function logic

**Review by axifive — APPROVED** [review]

**Pull request merged**